### ISSUE TITLE
Mysqli connection with SSL: Not working because of wrong socket/flags

### ISSUE THREAD

**Issue by Mecanik** [ADOdb/ADOdb#919]
I've been wasting quite some time trying to make this work, after debugging everything and making sure that connection itself works from any client -> server.

So the following snippet works fine:

```
$connectionID = mysqli_init();

if (!$connectionID) {
	die('mysqli_init failed');
}

if (!mysqli_options($connectionID, MYSQLI_INIT_COMMAND, 'SET AUTOCOMMIT = 0')) {
	die('Setting MYSQLI_INIT_COMMAND failed');
}

if (!mysqli_options($connectionID, MYSQLI_OPT_CONNECT_TIMEOUT, 5)) {
	die('Setting MYSQLI_OPT_CONNECT_TIMEOUT failed');
}

mysqli_ssl_set($connectionID, '...key', '...cert', '...ca', null, null);

if (!mysqli_real_connect($connectionID, 'hostname', 'username', 'password', 'database', 3306, null, MYSQLI_CLIENT_SSL)) {
	die('Connect Error (' . mysqli_connect_errno() . ') ' . mysqli_connect_error());
}

echo 'Success... ' . mysqli_get_host_info($connectionID) . "\n";

mysqli_close($connectionID);
```

However the following does not work at all:

```
$db = newAdoConnection('mysqli');

$db->ssl_key = '...key';
$db->ssl_cert = '...cert';
$db->ssl_ca = '...ca';
$db->ssl_capath = null;
$db->ssl_cipher = null;

$db->debug = true;

$db->Connect( 'hostname', 'username', 'password', 'database');
```

The error description is nothing information. Just "could not connect to..." and stack trace. 

So after checking the following: https://github.com/ADOdb/ADOdb/blob/11d361bf7f19fb1709b4e0fdfba9ccd171adb409/drivers/adodb-mysqli.inc.php#L200

I noticed the socket and flags are in the wrong order here. As per the PHP [documentation](https://www.php.net/manual/en/mysqli.real-connect.php) it's in the wrong place: 

```
socket
Specifies the socket or named pipe that should be used.

Note: Specifying the socket parameter will not explicitly determine the type of connection to be used when connecting to the MySQL server. How the connection is made to the MySQL database is determined by the host parameter.

flags
With the parameter flags you can set different connection options:

Supported flags
Name	Description
MYSQLI_CLIENT_COMPRESS	Use compression protocol
MYSQLI_CLIENT_FOUND_ROWS	return number of matched rows, not the number of affected rows
MYSQLI_CLIENT_IGNORE_SPACE	Allow spaces after function names. Makes all function names reserved words.
MYSQLI_CLIENT_INTERACTIVE	Allow interactive_timeout seconds (instead of wait_timeout seconds) of inactivity before closing the connection
MYSQLI_CLIENT_SSL	Use SSL (encryption)
MYSQLI_CLIENT_SSL_DONT_VERIFY_SERVER_CERT	Like MYSQLI_CLIENT_SSL, but disables validation of the provided SSL certificate. This is only for installations using MySQL Native Driver and MySQL 5.6 or later.
```

So to confirm this, I have edited my copy of the library to:

```
// SSL Connections for MySQLI
if ($this->ssl_key || $this->ssl_cert || $this->ssl_ca || $this->ssl_capath || $this->ssl_cipher) {
	mysqli_ssl_set($this->_connectionID, $this->ssl_key, $this->ssl_cert, $this->ssl_ca, $this->ssl_capath, $this->ssl_cipher);
	$this->socket = null;
	$this->clientFlags = MYSQLI_CLIENT_SSL;
}
```

And voila... connection works. I guess this begs the question... how could something so small and basic even happen? It's a library with 77 contributors and almost 2000 commits... does nobody read documentations?

This is just frustration beyond the point of being hilarious...

All I can say this is the last time/project I will ever use AdoDB. Better off writing a simple class of your own...

**Comment by mnewnham** [comment]
Are you telling me that open source software [sometimes has bugs](https://www.cisa.gov/uscert/apache-log4j-vulnerability-guidance)? I'm shocked, shocked, I tell you. I'm further shocked that a rarely used MySQL feature has sub-optimal testing. 

Anyway, thanks for the bug report, and off you go to write your class.

**Comment by dregad** [comment]
Fixed in be251d3ff570110b13f6b9577a372c5898980e19
